[lldb-dap] Add progress events to the packet list

JDevlieghere opened the pull request:
Before #134048, TestDAP_Progress relied on wait_for_event to block until
the progressEnd came in. However, progress events were not added to the
packet list, so this call would always time out. This PR makes it so
that packets are added to the packet list, and you can block on them.

Jlalond commented on lldb/test/API/tools/lldb-dap/progress/TestDAP_Progress.py:
```diff
@@ -12,17 +12,31 @@
 
 
 class TestDAP_progress(lldbdap_testcase.DAPTestCaseBase):
+    MAX_ATTEMPS = 10
```
Because we're in effect building a new state machine, wouldn't it make sense to add this to the underlying test case base? I was under the impression that `wait for event` would return quickly if progress already existed. I think having a function in the test base for this would be very useful. Plus, the underlying class leverages some concurrency primitives that we could reuse correct? 

If you don't want to implement that, you can always kick it back to me to move this logic into the base class.

JDevlieghere replied:
Yeah, that's a good point. Here's my understanding why how this works:

We have a packet read thread (`read_packet_thread`) which looks at incoming packets. Certain packets are recognized and added to lists (e.g. the `progress_events`) and they don't get added to the packets list, which means you can't wait on them:

```
            elif event.startswith("progress"):
                # Progress events come in as 'progressStart', 'progressUpdate',
                # and 'progressEnd' events. Keep these around in case test
                # cases want to verify them.
                self.progress_events.append(packet)
                # No need to add 'progress' event packets to our packets list.
                return keepGoing
```

If we wanted to make this work with `wait_for_event`, we can remove the last two lines and that way, we'll block until the packet is received. I actually like that approach better. Let me update the PR.

Jlalond reviewed (approved):
LGTM Round 2, left a comment this time on unifying this logic into the test base (Because I thought this is what the wait call already did!)

JDevlieghere committed:
[lldb-dap] Add progress events to the packet list

Before #134048, TestDAP_Progress relied on wait_for_event to block until
the progressEnd came in. However, progress events were not added to the
packet list, so this call would always time out. This PR makes it so
that packets are added to the packet list, and you can block on them.

JDevlieghere commented:
@Jlalond changed the approach and the description. Please take another look.

Jlalond reviewed (approved):
Much improved! I should've noticed this in my first patch but great fix for the flakiness
